sips/mev.md

```diff
@@ -0,0 +1,87 @@
+
+Currently, operators wishing to offer competitive returns must be sophisticated block builders with private transaction flows. This SIP aims to change that by allowing them to register with and propose blocks from external builders who comply to [ethereum/builder-specs](https://github.com/ethereum/builder-specs).
+
+## Specification
+
+### MEV-supporting operators
+
+Operators who wish to propose blinded blocks must configure their node to do so, otherwise it will propose standard locally-built blocks.
+
+### Validator registration
+
+Builders require validators to publish a [`builder-specs/SignedValidatorRegistration`](https://ethereum.github.io/builder-specs/#model-SignedValidatorRegistration) to set their `fee_recipient` and `gas_limit` preferences.
+
+When building a block for a validator, builders refer only to the registration with the highest timestamp.
+
+In the wild, some Ethereum validator clients currently produce and publish this message every epoch.
+
+#### Fee recipients
+
+Validators may set their preferred `fee_receipient` address by calling `setFeeRecipientAddress` in the `SSVNetwork` contract. Validators may repeat this call as their preference changes over time.
+
+#### Signing
+
+At the start of every slot, operators select their active validators with `ShouldRegisterValidatorAtSlot`, and for those, produce a [`builder-specs/SignedValidatorRegistration`](https://ethereum.github.io/builder-specs/#model-SignedValidatorRegistration) with their preferred `fee_recipient`.
+
+```go
+ValidatorRegistrationSlotInterval = 16 * SlotsPerEpoch
+
+func ShouldRegisterValidatorAtSlot(index phase0.ValidatorIndex, slot phase0.Slot) bool {
+    return (index + slot) % ValidatorRegistrationSlotInterval == 0
+}
+```
+
+#### Publishing
+
+At the end of every slot, operators publish registrations for the validators selected for signing in that slot, using their most recent successfully signed registration.
+
+> Note: Since relays/builders can have some delay with registrations, producing a blinded block before and shortly thereafter publishing the first registration may result in either:
+>
+> 1. Builder doesn't build a block, and Beacon node falls back to locally-built block from it's execution layer.
+> 2. Builder doesn't reward the validator's `fee_recipient` because it isn't aware of it yet.
+> 3. Builder rewards a potentially different `fee_recipient` from the validator's latest registration (such as a registration prior to onboarding to SSV.)
+
+#### Issue: Gas limits
+
+Unlike standard validator clients, gas limits are not set by validators, but rather by their operators.
+
+This SIP proposes to hardcode the gas limit to 30 million (which is the default in Prysm and Lighthouse), but recommends to keep watching it and modify if necessary.
```

Comment:
set where?

Reply:
In validator clients, it's configurable with a default of 30 million. In SSV it's currently hardcoded to the same default: https://github.com/bloxapp/ssv/blob/38193908b145dbe29186de34a6c60df1085c92e4/beacon/goclient/proposer.go#L27

Reply:
Do you think it should be configurable in SSV? @alonmuroch

Reply:
That's ethereum dependent so not sure

Reply:
We can let operators configure this, but if they differ, they will not be able to reconstruct signatures for `ValidatorRegistration` because it contains this value.

Social consensus on gas limit hasn't changed (at least) since The Merge, and if it does we'd want operators to switch at roughly the same time to avoid  differing `ValidatorRegistration`, so maybe waiting for this to change and then releasing a version with a transition scheduled at a specific epoch/slot is safer?

Reply:
Updated: https://github.com/bloxapp/SIPs/pull/25/files#diff-ffa7094aebeadab838e68f9af7d1be31bb06e525d5b9436cf8732dff04ab62bbR17-R19